### DISCUSSION TITLE
PHPMailer Website

### DISCUSSION THREAD

**Discussion by ghost** [PHPMailer/PHPMailer#3100] [Show and tell]
Hi there,
I have created a website for PHPMailer, so you can directly use this:

```php
require 'http://PHPMailer.rf.gd/src/Exception.php';
require 'http://PHPMailer.rf.gd/src/PHPMailer.php';
require 'http://PHPMailer.rf.gd/src/Exception.php/SMTP.php';

**Comment by Synchro** [comment]
I don't know quite how many ways there are to say that you should not do this. It's a truly terrible idea.

**Reply by ghost** [reply]
Why?

**Reply by Synchro** [reply]
All of these files are already available directly from here. What you suggest is simply asking to be turned into a supply-chain attack vector, it's unlikely to be kept up to date, the domain you're hosting on doesn't even have a valid TLS certificate and consequently is vulnerable to tampering in transit. Performance wise, you'd be asking it to do 3 HTTP requests every time your script is run, and these are also subject to failure, delay, firewalling, DNS issues, etc. It won't work on servers that don't have outbound internet access. Dependencies should be loaded using composer, not directly like this, as it means that if the classes ever change in their own dependencies, your script will break. I'm sure I could think of more reasons, but any one of these reasons is enough to not even consider it. In short:

- It's dangerous
- It's slow
- It's unreliable
- It's vulnerable
- It's not upgrade-resistant

Really, don't do this.

**Comment by ghost** [comment]
I understand these reasons, and I have taken the site down. (I could have added TLS, but I decided not to at the time)
Also, my only reason why I made it, was because I (and others) are using infinityfree to host there website, and we can't install it because we can't control the server.

**Reply by Synchro** [reply]
I don't understand that reasoning. If you can host a script that contains this code, you can use composer locally and upload everything. This is pretty normal for shared hosting accounts with no shell access.
